vespajlib/src/main/java/com/yahoo/text/Text.java

```diff
@@ -50,9 +50,10 @@ public static boolean isTextCharacter(int codepoint) {
 
         return (codepoint < 0x80)
                 ? allowedAsciiChars[codepoint]
-                : isTextCharAboveUsAscii(codepoint);
+                : (codepoint <  Character.MIN_SURROGATE) || isTextCharAboveMinSurrogate(codepoint);
```

Comment:
Doesn't this now fail all leading surrogates? They're `> MIN_SURROGATE` and `< MAX_HIGH_SURROGATE`, but what you _want_ to check is that it's either not a leading surrogate, or that it is a leading surrogate followed by a trailing one.